tests/test_flows.py

```diff
@@ -62,3 +63,23 @@ def test_divergence(stencil, dtype_device):
     print(divergence)
     assert (flow.ic_energy == pytest.approx(lattice.convert_to_numpy(ekin),rel=1))
     assert (0 == pytest.approx(divergence, abs=1e-4))
+
+@pytest.mark.parametrize("stencil", [D2Q9, D3Q27])
+def test_obstacle(stencil, dtype_device):
+    dtype, device = dtype_device
+    lattice = Lattice(stencil, dtype=dtype, device=device)
+
+    if stencil is D2Q9:
+        mask = np.zeros([20, 10])
+        mask[3:6, 3:6] = np.ones([3, 3]) > 0
+        flow = Obstacle2D(20, 10, 100, 0.1, lattice=lattice, char_length_lu=3)
+    if stencil is D3Q27:
+        mask = np.zeros([20, 10, 5])
+        mask[3:6, 3:6, :] = np.ones([3, 3, 5]) > 0
+        flow = Obstacle3D(20, 10, 5, 100, 0.1, lattice=lattice, char_length_lu=3)
+    collision = BGKCollision(lattice, tau=flow.units.relaxation_parameter_lu)
+    mask = torch.tensor(mask, dtype=torch.bool)
```

Comment:
Please make sure that this works with numpy masks. The basic philosophy here is that flows are defined in numpy while everything in the simulation uses torch.